### PR TITLE
issue 194 - pushing for a requested pull request

### PR DESCRIPTION
**Pull request by mhughes27**

### CODE REVIEW

File: app/models/award.rb [hunk 1]
```diff
@@ -21,4 +21,9 @@ def needs_role_input?
     award_type == "acting" 
   end
 
+  def pulldown_options
+    options_array = [['ACTING', 'acting'],['DIRECTING','directing'],['TECHNICAL','technical'],['PRODUCTION','production'],['ENSEMBLE','ensemble']]
```
Comment (sumeetjain) [inline app/models/award.rb]: Use `%w` or `%W` for an array of words.

Comment (sumeetjain) [inline app/models/award.rb]: Space missing after comma.

Comment (sumeetjain) [inline app/models/award.rb]: Line is too long. [146/80]

File: app/models/award.rb [hunk 2]
```diff
@@ -21,4 +21,9 @@ def needs_role_input?
...
+    return options_array
+  end
+
```
Comment (sumeetjain) [inline app/models/award.rb]: Extra empty line detected at class body end.

File: app/models/award.rb [hunk 3]
```diff
@@ -21,4 +21,11 @@ def needs_role_input?
     award_type == "acting" 
   end
 
+  def pulldown_options
```
Comment (sumeetjain) [inline app/models/award.rb]: Doesn't depend on instance state (maybe move it to another class?) (UtilityFunction)

File: app/models/award.rb [hunk 4]
```diff
@@ -21,4 +21,11 @@ def needs_role_input?
...
+    keys = %w[ACTING DIRECTING TECHNICAL PRODUCTION ENSEMBLE]
```
Comment (sumeetjain) [inline app/models/award.rb]: `%w`-literals should be delimited by `(` and `)`.

File: app/models/award.rb [hunk 5]
```diff
@@ -21,4 +21,11 @@ def needs_role_input?
...
+    values = %w[acting directing technical production ensemble]
```
Comment (sumeetjain) [inline app/models/award.rb]: `%w`-literals should be delimited by `(` and `)`.

File: app/models/award.rb [hunk 6]
```diff
@@ -21,4 +21,11 @@ def needs_role_input?
...
+    zipped = keys.zip(values)
+    return zipped
+  end
+
```
Comment (sumeetjain) [inline app/models/award.rb]: Extra empty line detected at class body end.

**Comment by mhughes27** [comment]
This original code worked, but was asked to extract it into a method:

<%=f.select(:award_type, options_for_select([['ACTING', 'acting'],['DIRECTING','directing'],['TECHNICAL','technical'],['PRODUCTION','production'],['ENSEMBLE','ensemble']])) %>

I created this method  in the Award class:

def pulldown_options
    keys = %w(ACTING DIRECTING TECHNICAL PRODUCTION ENSEMBLE)
    values = %w(acting directing technical production ensemble)
    zipped = keys.zip(values)
    return zipped
  end

It returns the array that is needed in the select_for_options

I called the method in the views

<%=f.select(:award_type, options_for_select(award.pulldown_options)) %>

It returns an undefined method error.

**Comment by sumeetjain** [comment]
Can you paste in the full error text?

**Comment by mhughes27** [comment]
I cannot right now because of a pending migration error.  My two branches created the same column in the awards table.

**Comment by sumeetjain** [comment]
That'll take a little in-person help to get through. Can you do this, for now:
1. Delete the duplicate migration file.
2. Reset your database:
   - `bundle exec rake db:drop`
   - `bundle exec rake db:create`
3. Then run migrations again: `bundle exec rake db:migrate`

**Comment by mhughes27** [comment]
NameError in Admin::Awards#new
Showing /Users/Hughes/Code/tag-awards/app/views/admin/awards/_form.html.erb where line #41 raised:

undefined local variable or method `award' for #<#Class:0x007ffabc7e9ac8:0x007ffabb439de8>
Extracted source (around line #41):
39
40
41
42
43
44

```
<div class="field-unit">
  <%= f.label :award_type %>
  <%=f.select(:award_type, options_for_select(award.pulldown_options)) %>
</div>

<div class="form-actions">
```

Trace of template inclusion: app/views/admin/awards/new.html.erb

Rails.root: /Users/Hughes/Code/tag-awards

Application Trace | Framework Trace | Full Trace
app/views/admin/awards/_form.html.erb:41:in `block in _app_views_admin_awards__form_html_erb__1861550406375817198_70357401577840'
app/views/admin/awards/_form.html.erb:17:in`_app_views_admin_awards__form_html_erb__1861550406375817198_70357401577840'
app/views/admin/awards/new.html.erb:40:in `_app_views_admin_awards_new_html_erb___1741726541036778845_70357401696440'

**Comment by sumeetjain** [comment]
Gotcha. So the error says that `award` is undefined. So the question to you then is: Where do you think `award` is being defined? (Link me to a specific file and cite the line number where it's being defined.)

**Comment by mhughes27** [comment]
I was trying to call a class method  - I capitalized the A in Award.  But now I get the error: undefined method `pulldown_options' for #Class:0x007ffabd130438

The method pulldown_options is defined in app/model/award.rb on line #24.

**Comment by sumeetjain** [comment]
Class methods should be defined like so:

``` ruby
def self.banana
  # method body goes here
end
```

See if that helps!

**Comment by mhughes27** [comment]
It works

**Pull request merged**